### PR TITLE
Opt out of "HTTPS everywhere": Do not warn http sources in source command when allowInsecureConnections is set to true

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes:https://github.com/NuGet/Home/issues/12791

Regression? Last working version:

## Description
Suppress the Http warning in source command when allowInsecureConnections is set to true.

Note: only `disable`, `enable`, `list` would be able to apply allowInsecureConnections.
For `add` and `update`, the `allowInsecureConnections` will always be false, the default value. As we do not accept the additional properties when add or update source (same with property protocalVersion). 
Based on our [discussion](https://teams.microsoft.com/l/message/19:994f8ef52dd948039177020177583736@thread.v2/1690499497810?context=%7B%22contextType%22%3A%22chat%22%7D), adding -allowInsecureConnections option to nuget sources add/update commands could be done later.

BTW, we have a PR about adding -protocolVersion option to nuget sources add/update commands in https://github.com/NuGet/NuGet.Client/pull/5002
We could probably add allowInsecureConnections option to nuget sources add/update commands as well.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by jeffkl — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs [hunk 1]
```diff
@@ -366,6 +503,76 @@ public void Sources_NoWarnWhenDisableHttpSource()
...
 
+        [Theory]
+        [InlineData("http://source.test", "true")]
+        [InlineData("http://source.test", "TRUE")]
+        [InlineData("http://source.test", "FALSE")]
+        [InlineData("http://source.test", "invalidString")]
+        [InlineData("http://source.test", "")]
+        [InlineData("https://source.test", "TRUE")]
+        [InlineData("https://source.test", "false")]
+        public void Sources_DisableHttpSourceWithAllowInsecureConnections_NoWarns(string source, string allowInsecureConnections)
```
Comment (aortiz-msft) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs]: For this test, I wonder if there's value in testing anything other than [InlineData("http://source.test", "false")]. 

@zivkan, @nkolev92 -  thoughts?

**Review by aortiz-msft — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs (hunk 1 above)
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs]: I agree. 

A lot of this is just testing the parsing of the allowinsecureconnections .  All of that should be unit testable. 

Also https://github.com/NuGet/Client.Engineering/issues/133. 

All of these tests are really really slow. 
I'm guessing this PR adds over a minute of test time altogether. 

I don't see why we need to test every combination.

**Review by nkolev92 — CHANGES_REQUESTED** [review]
I think the test are overly enthusiastic. 
They're testing so many different combinations, but in the end, in the product code, we only have a boolean. So we're effectively testing boolean parsing in a lot of these cases. 

I'm guessing there are plenty of unit tests regarding the parsing of `AllowInsecureConnections` from the config. 

We don't need to duplicate that here.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs (hunk 1 above)
Comment (heng-liu) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs]: Thanks for reviewing! Makes sense. I removed several tests.

**Comment by heng-liu** [comment]
Thanks for all the reviews! I removed some unnecessary tests. Pls take another look. Thanks!

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs [hunk 2]
```diff
@@ -156,6 +157,78 @@ public void SourcesCommandTest_EnableSource_WarnWhenUsingHttp()
             }
         }
 
+        [Theory]
+        [InlineData("http://source.test", "true", false)]
+        [InlineData("http://source.test", "false", true)]
```
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs]: I think even this is redundant. I'm guessing there's already a test that asserts a warning is raised...when no disabled. 

Same feedback applied.

Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs]: Thanks! Yes, there are tests testing about enabling and disabling http source already.
I removed the two newly added tests entirely.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs [hunk 3]
```diff
@@ -303,6 +359,82 @@ public void Sources_WarnWhenEnableHttpSource()
             }
         }
 
+        [Theory]
+        [InlineData("http://source.test", "true", false)]
```
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs]: Same as the other tests?

Comment (heng-liu) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs]: Yes.. sorry, just removed those tests and checked no newly added tests testing enabling and disabling http source.

**Review by aortiz-msft — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**